### PR TITLE
Added some information about coloring blocks and items

### PR DESCRIPTION
**Pull request by Nullcaller**
There's not so much to say on the subject, so I tried to make it compact and informative. I could write more about the JSON models, but it is not the section about block/item models, isn't it?

### CODE REVIEW

File: docs/blocks/blocks.md [hunk 1]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+----------------
+
+For some blocks, Minecraft paints one texture with different colors. Color of grass, vines, waterlilys and what not changes depending on the biome it's placed in.
+
+### Block Color Handlers
+
+Block color handler is just an instance of `IBlockColor`. The interface itself contains only one method. Its full description follows.
+
+```java
+colorMultiplier(
```
Comment (elifoster) [inline docs/blocks/blocks.md]: This isn't the full thing. Add its return type.

File: docs/blocks/blocks.md [hunk 2]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+For some blocks, Minecraft paints one texture with different colors. Color of grass, vines, waterlilys and what not changes depending on the biome it's placed in.
+
+### Block Color Handlers
+
+Block color handler is just an instance of `IBlockColor`. The interface itself contains only one method. Its full description follows.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: I would reword this as

"A block color handler is an instance of `IBlockColor`, which adds the `colorMultiplier` method below."

or something.

File: docs/blocks/blocks.md [hunk 3]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
 
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+For some blocks, Minecraft paints one texture with different colors. Color of grass, vines, waterlilys and what not changes depending on the biome it's placed in.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: "paints" is a weird word in this context.

Comment (elifoster) [inline docs/blocks/blocks.md]: "Color of grass..." -> "For example, the color of grass, vines, waterlilies, etc. change depending on the biome they are placed in."

File: docs/blocks/blocks.md [hunk 4]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+
+```java
+colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+It provides some information about the block, making it possible to change color dynamically. And if three first parameters are frequently met and therefore easy to understand, the last one causes all the trouble.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: "It provides... dynamically." -> "The objects passed to this method allow for the color multiplier of the block to be changed more dynamically."

File: docs/blocks/blocks.md [hunk 5]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+
+### Block Color Handlers
+
+Block color handler is just an instance of `IBlockColor`. The interface itself contains only one method. Its full description follows.
+
+```java
+colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos,
```
Comment (elifoster) [inline docs/blocks/blocks.md]: When and why this can be null needs to be explained.

File: docs/blocks/blocks.md [hunk 6]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+For some blocks, Minecraft paints one texture with different colors. Color of grass, vines, waterlilys and what not changes depending on the biome it's placed in.
+
+### Block Color Handlers
+
+Block color handler is just an instance of `IBlockColor`. The interface itself contains only one method. Its full description follows.
+
+```java
+colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn,
```
Comment (elifoster) [inline docs/blocks/blocks.md]: When and why this can be null needs to be explained.

File: docs/blocks/blocks.md (hunk 4 above)
Comment (elifoster) [inline docs/blocks/blocks.md]: "And if three first parameters are frequently met and therefore easy to understand, the last one causes all the trouble." makes no sense to me.

File: docs/blocks/blocks.md [hunk 7]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+It provides some information about the block, making it possible to change color dynamically. And if three first parameters are frequently met and therefore easy to understand, the last one causes all the trouble.
+
+Tint index is specified in block's model JSON file and literally is a number you set yourself just for you. Set multiple tint indexes and return different colors based on them, if you need two or more colors in your block. Block's digging and hitting particles are hardcoded to have zeroth tint index.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: "Tint index" -> "The tint index"

Comment (elifoster) [inline docs/blocks/blocks.md]: None of this line makes sense.

File: docs/blocks/blocks.md (hunk 1 above)
Comment (elifoster) [inline docs/blocks/blocks.md]: The return value needs to be explained in this section somewhere.

File: docs/blocks/blocks.md [hunk 8]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+
+### Registering a Block Color Handler
```
Comment (elifoster) [inline docs/blocks/blocks.md]: I don't think either of the `!!! important`s need to be `!!! important` here.

Instead, I would have the description be:
"Block color handlers are registered by calling `BlockColors#registerBlockColorHandler(IBlockColor, Block)`. A `BlockColors` instance can be obtained by calling `Minecraft#getBlockColors()`.

This must be done during the initialization phase, and on the client side only."

File: docs/blocks/blocks.md [hunk 9]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+
+Block color handlers are registered by calling `BlockColors.registerBlockColorHandler(colorHandler, block)` on an instance of `BlockColors`. Same can be done for muliple blocks.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: Change the code bit to `BlockColors#registerBlockColorHandler(IBlockColor, Block)`

File: docs/blocks/blocks.md [hunk 10]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+
+!!! importamt
+    This must be done during the initialization phase and only on the client side.
+
+!!! important
+    Instance of `BlockColors` is obtained via `Minecraft.getMinecraft().getBlockColors()`. Do NOT create any own instances, they will be absolutely useless.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: Change the code bit to `Minecraft#getBlockColors()`.

File: docs/blocks/blocks.md [hunk 11]
```diff
@@ -35,6 +35,41 @@ Blocks must be [registered][registering] to function.
...
+	
+### Registering a Handler for `ItemBlock`
+
+Instances of `ItemBlock` are items, and so they must be [registered and painted][] as items are.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: "painted" is a weird word to use in this context.

File: docs/items/items.md [hunk 12]
```diff
@@ -30,4 +30,33 @@ Registering an Item
 
 Items must be [registered][registering] to function.
 
-[registering]: ../concepts/registries.md#registering-things
+Coloring an Item
+----------------
+
+There are items Minecraft prefers to render by coloring one single texture. Leather helmets, spawn eggs, potions and other.
```
Comment (elifoster) [inline docs/items/items.md]: This doesn't make sense to me.

File: docs/items/items.md [hunk 13]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+### Item Color Handler
```
Comment (elifoster) [inline docs/items/items.md]: Pluralize "Handler" to "Handlers" to match with the "Block Color Handlers" section in the other file.

File: docs/items/items.md [hunk 14]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+Item color handler is just an instance of `IItemColor`. The interface contains only one method:
```
Comment (elifoster) [inline docs/items/items.md]: I would reword this to match my suggestion for the block one:

"An item color handler is an instance of `IItemColor`, which adds the `getColorFromItemStack` method as described below."

File: docs/items/items.md [hunk 15]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+```java
+getColorFromItemstack(
```
Comment (elifoster) [inline docs/items/items.md]: Add return type to this.

Comment (elifoster) [inline docs/items/items.md]: Explain the return value.

File: docs/items/items.md [hunk 16]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+  ItemStack stack, 
+  int tintIndex)
+```
+
+It provides ItemStack of item that is being painted and the tint index.
```
Comment (elifoster) [inline docs/items/items.md]: "painted" is a weird word in this context.

File: docs/items/items.md [hunk 17]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+```java
+getColorFromItemstack(
+  ItemStack stack, 
+  int tintIndex)
+```
+
+It provides ItemStack of item that is being painted and the tint index. 
+
+When coloring an ItemBlock, the tint index is the same thing as it is with blocks, but if an Item is being painted, the tint index value indicates which layer of an item is being painted. Layers of item are specified in its model JSON file and are layers of its texture.
```
Comment (elifoster) [inline docs/items/items.md]: "ItemBlock" -> "`ItemBlock`"

Comment (elifoster) [inline docs/items/items.md]: I'd reword this (and remove all the uses of "painted") to something along the lines of

```markdown
The tint index parameter has a different meaning depending on whether the item being colored is an `ItemBlock`.

For an `ItemBlock`, the tint index is exactly the same as it would be for its [block variant](../blocks/blocks.md#coloring-a-block).

For other kinds of items, the tint index indicates which texture layer is currently being drawn. These layers have the same meaning as they do in the item's model JSON file.
```

File: docs/items/items.md [hunk 18]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+### Registering an Item Color Handler
```
Comment (elifoster) [inline docs/items/items.md]: I don't think either of the `!!! important`s need to be `!!! important` here.

Instead, I would have the description be:
"Item color handlers are registered by calling `ItemColors#registerItemColorHandler(IItemColor, Item)`. An `ItemColors` instance can be obtained by calling `Minecraft#getItemColors()()`.

This must be done during the initialization phase, and on the client side only."

File: docs/items/items.md [hunk 19]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+Item color handlers are registered by calling `ItemColors.registerItemColorHandler(colorHandler, item)` on an instance of `ItemColors`. Same can be done for multiple items.
```
Comment (elifoster) [inline docs/items/items.md]: Change the syntax here to `ItemColors#registerItemColorHandler(IItemColor, Item)`

File: docs/items/items.md [hunk 20]
```diff
@@ -30,4 +30,33 @@ Registering an Item
...
+
+!!! important
+    This must be done during the initialization phase and only on the client side.
+	
+!!! important
+    Instance of `ItemColors` is obtained via `Minecraft.getMinecraft().getItemColors()`. Do NOT create any own, since they will be absolutely useless.
```
Comment (elifoster) [inline docs/items/items.md]: Change the syntax here to `Minecraft#getItemColors()`

**Comment by Nullcaller** [comment]
Thank you for a rich feedback, @elifoster. Hope these changes will do.

File: docs/items/items.md [hunk 21]
```diff
@@ -30,4 +30,29 @@ Registering an Item
...
+int getColorFromItemstack(
+  ItemStack stack, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer.
+
+If the subject of coloring is a regular `Item`, the tint index will show an index of item's texture layer that is being drawn. Layers of item's texture are specified in its model JSON file.
+
+But if the subject of coloring is an `ItemBlock`, the tint index will work exactly the same [way as it works for blocks](../blocks/blocks.md#coloring-a-block).
```
Comment (williewillus) [inline docs/items/items.md]: how is an itemblock a special case? / why even mention them if they aren't?

File: docs/blocks/blocks.md [hunk 22]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
...
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer. 
+
+Objects passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method. 
+
+The tint indexes are specified in block's model JSON file and are used to identify the element, color for painting which is requested. Block's digging and hitting particles are hardcoded to have zeroth tint index.
```
Comment (williewillus) [inline docs/blocks/blocks.md]: re 1st: i'm not quite sure what "identify the element, color for painting" means..
re 2nd: are you sure? I'm pretty sure I've made blocks that kept their tint when being broken

File: docs/items/items.md [hunk 23]
```diff
@@ -30,4 +30,29 @@ Registering an Item
...
+
+```java
+int getColorFromItemstack(
+  ItemStack stack, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer.
+
+If the subject of coloring is a regular `Item`, the tint index will show an index of item's texture layer that is being drawn. Layers of item's texture are specified in its model JSON file.
```
Comment (williewillus) [inline docs/items/items.md]: this is only true if you use `builtin/generated`. you can have a custom model with your tint indices, and have a custom IItemColor that looks for them.

I would reword this section to say that the tint index specified in the JSON is passed to getColorFromItemstack, but for `builtin/generated` models, each layer automatically gets a tint index assigned.

File: docs/blocks/blocks.md (hunk 22 above)
Comment (Nullcaller) [inline docs/blocks/blocks.md]: @williewillus, I am sure. The only elements that are not colored are the elements that do NOT have a tint index. But the particles are hardcoded to have zeroth. A bit of code from ParticleDigging with this particular moment showed follows.

```java
protected void multiplyColor(@Nullable BlockPos p_187154_1_)
{
    int i = Minecraft.getMinecraft().getBlockColors().colorMultiplier(this.sourceState, this.worldObj, p_187154_1_, 0);
    this.particleRed *= (float)(i >> 16 & 255) / 255.0F;
    this.particleGreen *= (float)(i >> 8 & 255) / 255.0F;
    this.particleBlue *= (float)(i & 255) / 255.0F;
}
```

**Comment by Nullcaller** [comment]
Thank you for your suggestions, @williewillus. These changes should satisfy you.

File: docs/blocks/blocks.md [hunk 24]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
...
+int colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer. 
+
+Objects passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: "Objects" -> "The `IBlockState`, `IBlockAccess`, and `BlockPos`"

Comment (elifoster) [inline docs/blocks/blocks.md]: Though the BlockModelRenderer thing is useful, I don't think that provides enough context.

File: docs/blocks/blocks.md [hunk 25]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
...
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer. 
+
+Objects passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method. 
+
+Tint indexes are specified for faces of an element in block's model JSON file. When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored. Block's digging and hitting particles are hardcoded to have zeroth tint index.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: "When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored."

The first sentence is not necessary here. This could just be "A face without a tint index won't be colored, and therefore will not have its color handler called." or something

File: docs/items/items.md [hunk 26]
```diff
@@ -30,4 +30,27 @@ Registering an Item
...
+
+```java
+int getColorFromItemstack(
+  ItemStack stack, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer.
+
+Tint indexes are specified for faces of an element in item's model JSON file. When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored. Layer indexes are used as tint indexes for items, models of which inherit `builtin/generted` model.
```
Comment (elifoster) [inline docs/items/items.md]: "When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored."

The first sentence is not necessary here. This could just be "A face without a tint index won't be colored, and therefore will not have its color handler called." or something

Comment (elifoster) [inline docs/items/items.md]: "Layer indexes are used as tint indexes for items, models of which inherit `builtin/generted` model." -> "Layer indexes are used as tint indexes for items whose models inherit from the `builtin/generated` model."

File: docs/items/items.md [hunk 27]
```diff
@@ -30,4 +30,27 @@ Registering an Item
...
+
+### Registering an Item Color Handler
+
+Item color handlers must be registered by calling `ItemColors#registerItemColorHandler(IItemColor, Item)` to function. Same can be done for multiple items. An instance of `ItemColors` can be obtained by calling `Minecraft#getItemColors()`. This must be done during the initialization phase and only on the client side.
```
Comment (elifoster) [inline docs/items/items.md]: What does "Same can be done for multiple items." mean? I don't understand it.

Comment (elifoster) [inline docs/items/items.md]: New line before "This must be done..." to match with the block doc.

**Comment by elifoster** [comment]
It's definitely getting there, and this is super useful to have documented. Some stuff is still a bit unclear/weirdly worded.

File: docs/blocks/blocks.md [hunk 28]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
 
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+There can be blocks, textures of which are colored within the code, and in fact, there are a lot of such blocks. For instance, color of grass, vines, waterlilys and etc. changes depending on the biome it's placed in.
```
Comment (Nullcaller) [inline docs/blocks/blocks.md]: Well, maybe... I am having trouble with an introduction to this section. I think I shall change it to "Block's texture can be dynamically colored, and that is already done for many blocks" or something.

Comment (elifoster) [inline docs/blocks/blocks.md]: "Blocks' textures can be programmatically colored. Many Vanilla blocks utilize this functionality, such as ..." perhaps?

Comment (Nullcaller) [inline docs/blocks/blocks.md]: Yes, I think.

File: docs/blocks/blocks.md [hunk 29]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
...
+
+This method returns a hex representation of a color in an integer. 
+
+Objects passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method. 
+
+Tint indexes are specified for faces of an element in block's model JSON file. When a face with a tint index is being drawn, its tint index is passed to the method. A face without a tint index won't be colored. Block's digging and hitting particles are hardcoded to have zeroth tint index.
+
+### Registering a Block Color Handler
+
+Block color handlers must be registered by calling `BlockColors#registerBlockColorHandler(IBlockColor, Block)` to function. Same can be done for muliple blocks. An instance of `BlockColors` can be obtained by calling `Minecraft#getBlockColors()`.
```
Comment (Nullcaller) [inline docs/blocks/blocks.md]: It means that you can do something like this:
```java
Minecraft.getMinecraft.getBlockColors().registerBlockColorHandler(new IBlockColor {...}, blockA, blockB, blockC, blockD)
```

Comment (elifoster) [inline docs/blocks/blocks.md]: Oh. Well then its signature in the method reference should be changed to `BlockColors#registerBlockColorHandler(IBlockColor, Block...)` and that sentence can be removed.

File: docs/blocks/blocks.md [hunk 30]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
...
+
+This method returns a hex representation of a color in an integer. 
+
+The `IBlockState`, `IBlockAccess` and `BlockPos` passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method.
+
+Tint indexes are specified for faces of an element in block's model JSON file. A face without a tint index won't be colored, and therefore won't have its color handler called. Block's digging and hitting particles are hardcoded to have zeroth tint index.
+
+### Registering a Block Color Handler
+
+Block color handlers must be registered by calling `BlockColors#registerBlockColorHandler(IBlockColor, Block)` to function. Same can be done for muliple blocks. An instance of `BlockColors` can be obtained by calling `Minecraft#getBlockColors()`.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: Change the method reference to `BlockColors#registerBlockColorHandler(IBlockColor, Block...)`

File: docs/items/items.md [hunk 31]
```diff
@@ -30,4 +30,29 @@ Registering an Item
...
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer.
+
+Tint indexes are specified for faces of an element in item's model JSON file. A face without a tint index won't be colored, and therefore will not have its color handler called. Layer indexes are used as tint indexes for items whose models inherit from the `builtin/generted` model.
+
+### Registering an Item Color Handler
+
+Item color handlers must be registered by calling `ItemColors#registerItemColorHandler(IItemColor, Item)` to function. Same can be done for multiple items. An instance of `ItemColors` can be obtained by calling `Minecraft#getItemColors()`.
```
Comment (elifoster) [inline docs/items/items.md]: Change the method reference to

"`ItemColors#registerItemColorHandler(IItemColor, Item...)` (or `ItemColors#registerItemColorHandler(IItemColor, Block...)` for `ItemBlock`s)

File: docs/blocks/blocks.md [hunk 32]
```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
...
+int colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer. 
+
+The `IBlockState`, `IBlockAccess` and `BlockPos` passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method.
```
Comment (elifoster) [inline docs/blocks/blocks.md]: As I said in a previous review:


Though the BlockModelRenderer thing is useful, I don't think that provides enough context.

File: docs/items/items.md (hunk 31 above)
Comment (Nullcaller) [inline docs/items/items.md]: No, that's not right. `ItemBlock` extends **`Item`**, but not of the `Block`.

**Comment by Nullcaller** [comment]
There. It seems like all's done, if I didn't screw up the part with the `BlockModelRenderer` :)

File: docs/blocks/blocks.md [hunk 33]
```diff
@@ -35,6 +35,43 @@ Blocks must be [registered][registering] to function.
 
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+Blocks' textures can be programmatically colored. Many Vanilla blocks utilize this functionality. For instance, color of grass, vines, waterlilys and etc. changes depending on the biome it's placed in.
```
Comment (tterrag1098) [inline docs/blocks/blocks.md]: Either "A block's" or just "Block textures"

Comment (tterrag1098) [inline docs/blocks/blocks.md]: For instance, -> For instance:
changes -> change color
biome it's placed in -> biome they are in

File: docs/blocks/blocks.md [hunk 34]
```diff
@@ -35,6 +35,43 @@ Blocks must be [registered][registering] to function.
...
+  int tintIndex)
+```
+
+#### Return Value
+
+This method returns a hex representation of a color in an integer. 
+
+#### Parameters
+
+The `IBlockState`, `IBlockAccess` and `BlockPos` passed to the method make it possible to change color multiplier dynamically, but the method can be called with null `IBlockAccess` and `BlockPos` by `BlockModelRenderer`, when it renders model's brightness.
```
Comment (tterrag1098) [inline docs/blocks/blocks.md]: "when it renders model's brightness" what does this mean?

File: docs/blocks/blocks.md [hunk 35]
```diff
@@ -35,6 +35,43 @@ Blocks must be [registered][registering] to function.
...
+
+Tint indexes are specified for faces of an element in block's model JSON file. A face without a tint index won't be colored, and therefore won't have its color handler called. Block's digging and hitting particles are hardcoded to have zeroth tint index.
```
Comment (tterrag1098) [inline docs/blocks/blocks.md]: indexes -> indices
Block's digging -> Block digging (or just drop the word 'Block')

File: docs/items/items.md [hunk 36]
```diff
@@ -30,4 +30,33 @@ Registering an Item
 
 Items must be [registered][registering] to function.
 
-[registering]: ../concepts/registries.md#registering-things
+Coloring an Item
+----------------
+
+Items' textures can be programmatically colored. Many Vanilla items utilize this functionality. For example, leather helmets, spawn eggs, potions and other.
```
Comment (tterrag1098) [inline docs/items/items.md]: Same changes as above

File: docs/blocks/blocks.md (hunk 34 above)
Comment (Nullcaller) [inline docs/blocks/blocks.md]: Well, it calls the method with null second and third parameters from `renderModelBrightness`. I have no idea how to rephrase it.

Comment (tterrag1098) [inline docs/blocks/blocks.md]: In that case I don't think it's worth mentioning specifics. Just note that those parameters *are* nullable, not when.

**Comment by Nullcaller** [comment]
There.

**Comment by Nullcaller** [comment]
If you have no more improvement suggestions, it's done.
